include/h2o.h

```diff
@@ -156,6 +157,7 @@ typedef struct st_h2o_filter_t {
     void (*on_context_dispose)(struct st_h2o_filter_t *self, h2o_context_t *ctx);
     void (*dispose)(struct st_h2o_filter_t *self);
     void (*on_setup_ostream)(struct st_h2o_filter_t *self, h2o_req_t *req, h2o_ostream_t **slot);
+    void (*on_send_early_hints)(struct st_h2o_filter_t *self, h2o_req_t *req);
```

Comment:
We might want to rename it to `on_early_hints`. How it is being to be handled (e.g. "send") is what each protocol handler decides.